packages/web3-eth/src/errors.ts

```diff
@@ -29,15 +29,27 @@ export class InvalidTransactionWithSender extends Web3Error {
 	public code = ERR_TX_INVALID_SENDER;
 
 	public constructor(value: unknown) {
-		super(JSON.stringify(value), 'invalid transaction with sender');
+		super(
+			JSON.stringify(
+				value,
+				(_, v) => (typeof v === 'bigint' ? v.toString() : v) as unknown,
+			).replace(/['"]+/g, ''),
```

Comment:
I would suggest to move this logic of parsing data to string into `web3-utils` package and then use here.